### ISSUE TITLE
Add ARM binaries to releases

### ISSUE THREAD

**Issue by cor** [paritytech/polkadot#5730]
Currently, there are only X86 binaries available on [the releases page](https://github.com/paritytech/polkadot/releases/). This is a problem for our team as some people are developing on M1 Macs. Can you please also release ARM binaries?

**Comment by chevdor** [comment]
The reason why we release only X86 binaries is that validators usually don't run nodes on anything else than Linux servers.
That being said, many devs are using Intel Mac and Mx Mac machines but usually build the binaries they need.

When building a node is an annoyance (it takes a while) and devs just need to run a node, there is also the option to use the [Docker image](https://hub.docker.com/r/parity/polkadot/tags) we provide.

Could you provide details if your use case does not fall in the categories above ?

**Comment by coderobe** [comment]
Note that the linux x64 docker image should also run fine on m1 macs (tested with a current dev build of podman w/ qemu)

**Comment by shawntabrizi** [comment]
I just came back from the Parachains Summit, and there were others asking for ARM builds, specifically to speed up development, since you need a polkadot binary to run a test relay chain

**Comment by xlc** [comment]
AWS Graviton arm server is getting popular and it will be great if we can run polkadot nodes with it.

**Comment by kianenigma** [comment]
+1

**Comment by bkchr** [comment]
@chevdor compiling the binary for arm64 should work without any problems and it gets more and more popular as @xlc. Please prepare the necessary steps that we can have arm64 builds in the next release.

**Comment by johnmela** [comment]
Any idea when the next release containing the arm64 binaries will come out? Will it target `v0.9.30`?

**Comment by coderobe** [comment]
We are looking into it.

> Will it target `v0.9.30`?

no.

**Comment by andreclaro** [comment]
+1

**Comment by bredamatt** [comment]
+1

**Comment by ggwpez** [comment]
I think its being worked on here https://github.com/paritytech/polkadot/pull/6108

**Comment by chevdor** [comment]
Indeed, I am on the topic atm. If someone wants to test, feel free to ping me, I can provide an unofficial yet built in production profile for `polkadot v0.9.37`. For now I am testing it and working on the integration with our release process.
The PR mentioned above should also provide everything allowing to build such a binary already today.

**Comment by davedomainx** [comment]
@chevdor I am quite interested in testing out this arm64 image.  Can you please let me know how get this image (is there a docker image available?)

**Comment by chevdor** [comment]
There is no docker image yet.

I can provide a production build of an early RC of v0.9.38 for `aarch64-unknown-linux-gnu` but I don't have the latest/final build yet and no recent one for M1s.

The version linked below is:
- really only for testing
- do **NOT** run a validator on it, there are several reasons not to do so that I will cover when we publish those binaries
- **NOT** the final v0.9.38 (it was RC5 if I recall and we added several fixes to the final)

You can get the build here: https://github.com/chevdor/polkadot/releases/tag/v0.9.38-beta

That being said, I would definitely recommend **against** considering the arm64 binary for a validator. 
The binary itself is fine but I ran the benchmarks on various providers and machines and all the results were showing major  weaknesses related to the single core performance, making those unsuitable for a validator. Nothing near the `...performance gains of up to 36%`.

If you find a provider with good specs and price, please share your findings (I checked GCP and AWS).

Besides that, the build did work fine. I tested mostly wrap syncs with ParityDB and upgrading from v0.9.37 as well.
It all went fine and the nodes were running "solid" :)

**Comment by polkalegos** [comment]
Thanks for your work @chevdor. Do you plan to publish another build for M2/M2Pro?

**Comment by kogeler** [comment]
You can try to use [my scripts](https://github.com/kogeler/substrate-cross-compiler-env) to build an aarch64 binary that is optimized for your CPU. 

Of course they are not suitable for production builds.

**Comment by okalenyk** [comment]
Hi everyone! Any updates or approximate ETA on this? Found out that it would be great to have an ability to launch zombienet kubernetes configuration locally having the Mac M1 with Docker Desktop based kubernets that runs ARM64 node. Currently I was able to make changes to Parity [base-ci-linux](https://hub.docker.com/r/paritytech/base-ci-linux) and [ci-linux](https://hub.docker.com/r/paritytech/ci-linux) Dockerfile to build them for ARM64 and succeded.

Here are the links to images:

[base-ci-linux](https://hub.docker.com/r/okalenyk/base-ci-linux/tags) - required a single line of code to make it work cross-platform
[ci-linux](https://hub.docker.com/r/okalenyk/ci-linux/tags) - same thing here

I can provide PRs if someone interested - just don't want to litter parity github repos.

I don't recommend to use the in production for sure, but for local development might be helpful.

**Comment by chevdor** [comment]
There is some work probably interesing following in https://github.com/paritytech/ci_cd/issues/609
I am also trying to dissociate bin and docker images which require some work in other areas. More on that once it becomes usable.

Atm, if this is just to make it convenient, you could make a build or used one of those I published.
If this is for local dev, you may also skip the docker part entirely (unless you need/want it).

**Comment by frisitano** [comment]
> Hi everyone! Any updates or approximate ETA on this? Found out that it would be great to have an ability to launch zombienet kubernetes configuration locally having the Mac M1 with Docker Desktop based kubernets that runs ARM64 node. Currently I was able to make changes to Parity [base-ci-linux](https://hub.docker.com/r/paritytech/base-ci-linux) and [ci-linux](https://hub.docker.com/r/paritytech/ci-linux) Dockerfile to build them for ARM64 and succeded.
> 
> Here are the links to images:
> 
> [base-ci-linux](https://hub.docker.com/r/okalenyk/base-ci-linux/tags) - required a single line of code to make it work cross-platform [ci-linux](https://hub.docker.com/r/okalenyk/ci-linux/tags) - same thing here
> 
> I can provide PRs if someone interested - just don't want to litter parity github repos.
> 
> I don't recommend to use the in production for sure, but for local development might be helpful.

Could you share the code changes with me please?

**Comment by okalenyk** [comment]
> > Hi everyone! Any updates or approximate ETA on this? Found out that it would be great to have an ability to launch zombienet kubernetes configuration locally having the Mac M1 with Docker Desktop based kubernets that runs ARM64 node. Currently I was able to make changes to Parity [base-ci-linux](https://hub.docker.com/r/paritytech/base-ci-linux) and [ci-linux](https://hub.docker.com/r/paritytech/ci-linux) Dockerfile to build them for ARM64 and succeded.
> > Here are the links to images:
> > [base-ci-linux](https://hub.docker.com/r/okalenyk/base-ci-linux/tags) - required a single line of code to make it work cross-platform [ci-linux](https://hub.docker.com/r/okalenyk/ci-linux/tags) - same thing here
> > I can provide PRs if someone interested - just don't want to litter parity github repos.
> > I don't recommend to use the in production for sure, but for local development might be helpful.
> 
> Could you share the code changes with me please?

@frisitano,

Sure, here are the links:

- [*base-ci-linux docker multi platform support](https://github.com/paritytech/scripts/pull/541)
- [*ci-linux docker arm64 support](https://github.com/paritytech/scripts/pull/542)
